probe/proc/reader_linux.go

```diff
@@ -0,0 +1,203 @@
+package proc
+
+import (
+	"bytes"
+	"path"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/bluele/gcache"
+)
+
+const (
+	filesCacheLen        = 512
+	filesCacheExpiration = 60 * time.Second
```

Comment:
Just to be clear, a single file is for example /proc/<pid>/something, is that correct? If so, is 60 seconds perhaps too long?